Makefile

```diff
@@ -53,6 +38,9 @@ dev-dynamic-mem: dev-dynamic
 
 # test runs the unit tests and vets the code
 test: prep
```

Comment:
Unfortunately the circle job `test-go` doesn't invoke this make target, so is still going to skip api/sdk tests. That will require a similar expansion.

Reply:
I don't know what the `test-go` job is or if we use it, but I'm sure our release engineer can make it work.

Reply:
Not sure I understand. The circleici `test-go` job runs all tests apart than those in `./vendor` and `./integ`, so this change shouldn't impact that as far as I can see, unless I'm missing something @rboyer?

Reply:
Great, thanks for the explanation @rboyer. This only applies when GO111MODULE is on, which it should not be at the moment. I now realise that the change to turn that off hadn't made it to master yet, so I've fixed that by merging https://github.com/hashicorp/vault/pull/7703 which has solved the problem.

Just verified that api and sdk tests are being run on master now: https://circleci.com/gh/hashicorp/vault/24698 

Good to know that when we do move to Go modules (targetting 1.4/1.5) we'll need to address this then.